hazelcast/src/main/java/com/hazelcast/wan/impl/InternalWanEvent.java

```diff
@@ -16,17 +16,24 @@
 
 package com.hazelcast.wan.impl;
 
+import com.hazelcast.cache.impl.CacheService;
 import com.hazelcast.internal.serialization.Data;
 import com.hazelcast.internal.util.Clock;
+import com.hazelcast.map.impl.MapService;
 import com.hazelcast.wan.WanEvent;
+import com.hazelcast.wan.WanEventCounters;
+import com.hazelcast.wan.WanEventService;
 
 import javax.annotation.Nonnull;
 import java.util.Set;
 
 /**
- * Private API for WAN replication events
+ * Private API for WAN replication events. Adds methods for internal use by
+ * our built-in WAN replication implementation.
+ *
+ * @param <T> type of event data
```

Comment:
`@param <T> type of object on which the event occurred`?

event data send my mind to `Data`.

Reply:
True, but it's only because we live and breathe `Data` every day :) Not sure if an end-user will think of it the same way. Type of object on which the event occurred can also refer to an `IMap` in the case of map updates. I'll try to think of something.

Reply:
Changed to your suggestion, couldn't come up with anything else :)